flocker/node/agents/test/test_blockdevice.py

```diff
@@ -982,12 +983,13 @@ def test_no_devices_one_dataset(self):
             nodes={
                 Node(
                     uuid=uuid,
-                    hostname=node,
                     manifestations={dataset_id: manifestation},
                 )
             }
         )
-        state = DeploymentState(nodes=[])
+        state = DeploymentState(nodes=[NodeState(
+            uuid=uuid, hostname=node, applications=[], manifestations={},
+            devices={}, paths={})])
```

Comment:
This is still sort of invalid, right?  This `NodeState` represents `applications` but not `used_ports` which isn't something actually possible in real use.

We have `flocker.node.testtools.EMPTY` and `EMPTY_STATE`.  I wonder if we should have similar things for `Node` and `NodeState`.  `EMPTY_NODE_STATE`, `EMPTY_NODE_APP_STATE`, `EMPTY_NODE_DATASET_STATE`.  Centralize proper initialization of all these attributes.  File as a follow-up issue if you want.

Another thing that might be nice is more invariant enforcement on `NodeState`.  Knownness of `used_ports` and `applications` matching could be enforced; likewise for knownness of `manifestations`, `devices`, `paths`.  Another follow-up issue, probably.